pkg/controller/constraint/constraint_controller.go

```diff
@@ -338,20 +335,44 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 					}
 					currentVapBinding = nil
 				}
-				newVapBinding := &admissionregistrationv1beta1.ValidatingAdmissionPolicyBinding{}
+				newVapBinding := &unstructured.Unstructured{}
+				newVapBinding.SetGroupVersionKind(schema.GroupVersionKind{Group: groupVersion.Group, Version: groupVersion.Version, Kind: "ValidatingAdmissionPolicyBinding"})
 				transformedVapBinding, err := transform.ConstraintToBinding(instance, VAPEnforcementActions)
 				if err != nil {
 					status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
 					if err2 := r.writer.Update(ctx, status); err2 != nil {
-						log.Error(err2, "could not report transform vapbinding error status")
+						log.Error(err2, "could not report transform vapbinding error", "vapName", vapBindingName)
 					}
 					return reconcile.Result{}, err
 				}
 				if currentVapBinding == nil {
-					newVapBinding = transformedVapBinding.DeepCopy()
+					err = r.scheme.Convert(transformedVapBinding, newVapBinding, nil)
```

Comment:
Will convert honor the GVK set on the unstructured object? It may be better/simpler to have `newVapBinding` be of `Object` type (the generic interface that matches all K8s objects, and set it to an empty instance of the appropriate versioned object via a function.

ex:

```
newVapBinding, err := vapBindingForVersion(gvk) // do err check
r.scheme.Convert(transformedVapBinding, newVapBinding, nil)


func vapBindingForVersion(gvk scheme.GroupVersion) (runtime.Object, error) {
   switch gvk.Version {
   case "v1":
      return &v1.VapBinding{}
   ....
   default:
     return nil, errors.New("unrecognized version")
   }
}
```

Doing it this way will avoid uncertainties around how the type system handles unstructured and excess version juggling.